app/legacy_lib/import_stripe/subscription_to_card.rb

```diff
@@ -0,0 +1,70 @@
+# License: AGPL-3.0-or-later WITH Web-Template-Output-Additional-Permission-3.0-or-later
+
+class ImportStripe::SubscriptionToCard
+  include ActiveModel::AttributeAssignment
+
+  attr_accessor :nonprofit, :row, :copy_row
+
+  def initialize(attr = {})
+    assign_attributes(attr)
+  end
+
+  def amount
+    (row[:amount] * 100).to_i
+  end
+
+  def current_period_ends
+    Format::Date.parse(row[:current_period_end_utc])
+  end
+
+  def customer_name
+    row[:customer_description]
+  end
+
+  def customer_email
+    row[:customer_email]
+  end
+
+  def product
+    row[:product]
+  end
+
+  def stripe_card_id
+    copy_row[:source_id_new]
+  end
+
+  def stripe_customer_id
+    row[:customer_id]
+  end
+
+  def supporter
+    nonprofit.supporters.create_with(name: customer_name).find_or_create_by!(email: customer_email)
+  end
+
+  def card
+    supporter.cards.create_with(email: customer_email, name: "Imported Card").find_or_create_by!(stripe_customer_id:, stripe_card_id:)
+  end
+
+  def token
+    InsertSourceToken.create_record(card)
```

Comment:
Does this need to be memoized? (Can I call it many times and get the same result safely?)

Reply:
I don't think it's a problem but in retrospect, probably good to memoize